x-pack/legacy/plugins/lens/server/plugin.tsx

```diff
@@ -5,28 +5,51 @@
  */
 
 import { Server, KibanaConfig } from 'src/legacy/server/kbn_server';
-import { Plugin, CoreSetup, SavedObjectsLegacyService } from 'src/core/server';
+import { Plugin, CoreSetup, CoreStart, SavedObjectsLegacyService } from 'src/core/server';
 import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { Subject } from 'rxjs';
+import { first } from 'rxjs/operators';
+import {
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '../../../../plugins/task_manager/server';
 import { setupRoutes } from './routes';
-import { registerLensUsageCollector, initializeLensTelemetry } from './usage';
+import {
+  registerLensUsageCollector,
+  initializeLensTelemetry,
+  scheduleLensTelemetry,
+} from './usage';
 
 export interface PluginSetupContract {
   savedObjects: SavedObjectsLegacyService;
   usageCollection: UsageCollectionSetup;
   config: KibanaConfig;
   server: Server;
+  taskManager: TaskManagerSetupContract;
 }
 
+export interface PluginStartContract {
+  server: Server;
+  taskManager: TaskManagerStartContract;
+}
+
+const taskManagerStartContract$ = new Subject<TaskManagerStartContract>();
```

Comment:
I'm surprised that Lens is responsible here for getting an instance of the task manager. Why isn't this instantiated somewhere else?

Reply:
This is just because Lens is still a Legacy plugin.
Once Lens is migrated to NP, this should happen automatically.

Reply:
Looking at it again I realise there's another issue in Lens specifically - during the Setup stage It's trying to use a TaskManager API that's only at Start stage, so we need to wrap that in some async method of using it - that's what the Subject is doing.